### PR TITLE
`Development`: Improve setup instructions

### PR DESCRIPTION
**Pull request by pal03377**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context
I set up a local instance of Artemis and had some minor problems:
- There were slightly outdated instructions (e.g. changed configuration page names in Bitbucket).
- I had some minor undocumented issues which I had to figure out myself.

Although none of these issues made it impossible to set up Artemis locally, I wanted to provide some additional / changed instructions for added convenience of future developers.

### Description
I created notes about every problem I had while setting up a local instance of Artemis and changed the Setup documentation accordingly afterwards. While doing that, I also noticed a minor error in the Sphinx build instructions and fixed that as well. 

I updated a screenshot and changed the filename to be more consistent with the filenames of other screenshots in the same folder.

I did not modify any actual code.

https://artemis-platform--6091.org.readthedocs.build/en/6091/dev/setup/

### Steps for Testing

Prerequisites:
- Sphinx installation

1. Build the documentation (see `docs/README.md`)
2. Open the documentation page and navigate to the "Setup Guide".
3. Read through the changes.

#### Code Review
- [x] Review 1
- [x] Review 2

### CODE REVIEW

**Review by ge65cer — DISMISSED** [review]
Congratulations on your first contribution! 🎉  Thanks for the fixes, looks good to me!

File: docs/dev/setup/bamboo-bitbucket-jira.rst [hunk 1]
```diff
@@ -59,7 +59,8 @@ Configure Bamboo, Bitbucket and Jira
 
 By default, the Jira instance is reachable under ``localhost:8081``, the
 Bamboo instance under ``localhost:8085`` and the Bitbucket instance
-under ``localhost:7990``.
+under ``localhost:7990``. You will probably have to change the respective URLs in
+``application-local.yml`` (e.g. set ``artemis.version-control.url`` to ``http://localhost:7990``).
```
Comment (JohannesStoehr) [inline docs/dev/setup/bamboo-bitbucket-jira.rst:63]: As fast as I can see, they already have that name in application-artemis.yml and in the example in the server setup for the application-local.yml.
Only this location here uses the version without the `http://`

File: docs/dev/setup/bamboo-bitbucket-jira.rst [hunk 2]
```diff
@@ -73,7 +74,11 @@ under ``localhost:7990``.
    in all 3 applications.
    For the Bamboo database you can choose H2.
    Also, you can select the evaluation/internal/test/dev setups if you are asked.
-   Follow the additional steps for Jira and Bitbucket.
+
+   Put the admin username into ``application-local.yml`` at ``artemis.version-control.user``
```
Comment (JohannesStoehr) [inline docs/dev/setup/bamboo-bitbucket-jira.rst:78]:
```suggestion
   Put the admin username and password into ``application-local.yml`` at ``artemis.version-control.user``
```

File: docs/dev/setup/bamboo-bitbucket-jira.rst [hunk 3]
```diff
@@ -197,12 +202,15 @@ under ``localhost:7990``.
 
                 Adding Crowd Server in **Bamboo**
 
-#. Give the test users User access on Bitbucket: Configure → Global permissions
+#. Give the test users User access on Bitbucket: On the Administration interface (settings cogwheel on the top),
+   go to the Global permissions. Type the names of all test users in the search field ("Add Users) and give them
```
Comment (JohannesStoehr) [inline docs/dev/setup/bamboo-bitbucket-jira.rst:206]:
```suggestion
   go to the Global permissions. Type the names of all test users in the search field ("Add Users") and give them
```

File: docs/dev/setup/bamboo-bitbucket-jira.rst [hunk 4]
```diff
@@ -229,13 +237,13 @@ under ``localhost:7990``.
 
    #. Personal access token for Bamboo.
 
-      - Log in as the admin user and go to Bamboo -> Profile (top right corner) -> Personal access tokens ->
+      - Log in as the admin user and go to Bamboo → Profile (top right corner) → Personal access tokens →
```
Comment (JohannesStoehr) [inline docs/dev/setup/bamboo-bitbucket-jira.rst:240]: This arrow is used in some more places of the documentation (49 in total). I'd suggest going through them and replacing all of them

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
The file `application-local.yml` is only optional and it also works with only the `application-artemis.yml`. This is probably the reason of the different file names.
If we want to change it, it should be clearer during the normal server setup that the `application-local.yml` will be used in the next steps for e.g. bamboo or that the steps can be executed using `application-artemis.yml`

File: docs/dev/setup/bamboo-bitbucket-jira.rst (hunk 1 above)
Comment (pal03377) [reply docs/dev/setup/bamboo-bitbucket-jira.rst:63]: The default URLs in `application-artemis.yml` [are the ones from TUM](https://github.com/ls1intum/Artemis/blob/develop/src/main/resources/config/application-artemis.yml#L20). 

I will move this info to the part of the guide where admin username + password for Bitbucket/Jira/Bamboo are set, as the URLs are also changed there. When I followed the instructions, I think I didn't notice that and just changed the credentials in my local configuration.

File: docs/dev/setup/bamboo-bitbucket-jira.rst (hunk 4 above)
Comment (pal03377) [reply docs/dev/setup/bamboo-bitbucket-jira.rst:240]: I think I would rather open another PR about that later to keep the scope of this PR constrained to just the setup instructions.

**Comment by pal03377** [comment]
> The file `application-local.yml` is only optional and it also works with only the `application-artemis.yml`. This is probably the reason of the different file names. If we want to change it, it should be clearer during the normal server setup that the `application-local.yml` will be used in the next steps for e.g. bamboo or that the steps can be executed using `application-artemis.yml`

At the start, the current setup instructions say this:
> You can change the options directly in the file `application-artemis.yml` in the folder `src/main/resources/config`. However, you have to be careful that you do not accidentally commit your password. Therefore, we strongly recommend, to create a new file `application-local.yml` in the folder `src/main/resources/config` which is ignored by default. You can override the following configuration options in this file.

Because of this, I think it's best if the rest of the setup documentation consistently recommends changing the configuration in `application-local.yml` instead of switching between different files.

**Review by JohannesStoehr — APPROVED** [review]
Thanks for addressing my suggestions. Approve

**Review by ge65cer — APPROVED** [review]

**Review by simon-hng — APPROVED** [review]
Improves setup instructions and makes them more up-to date.
Thanks a lot!

**Pull request merged**